pkg/api/file.go

```diff
@@ -41,7 +41,7 @@ const (
 	EncryptHeader     = "swarm-encrypt"
 )
 
-type targetsContextKey struct{}
+type TargetsContextKey struct{}
```

Comment:
This type should not be exported, or not in the `api` package. This is a good example of a problematic design. By exporting it in `api` package, other packages that need this information must import `api` package creating a possibility for circular imports and breaking the layers of abstraction. Move this type to an appropriate package. Package `api` is a "top level" package that imports and depends other packages. This change makes `chunk` and `netstore` packages unimportable in this package.

I would even suggest not to export it as the purpose of the custom types for context keys is to avoid conflicts. Functions that would set and get required values with using unexported type for context key would be ideal, but not strictly required. Of course, not to define them in the `api` package.

Reply:
Yes, this stands out. I would suggest to move this to the `sctx` package, and use setters and getters on the context in the same manner it is used on the other values there